### DISCUSSION TITLE
How to pass in function params in a jsonb_insert? (supabase RPC)

### DISCUSSION THREAD

**Discussion by theljwhite** [supabase/supabase#10705] [Questions]
Hey, sorry if I'm an idiot for this, def not my area of expertise... I have a JSONB column, "character_quests" that I need to insert objects into.

It looks like this:

column - character_quests: 
`{
  "questsarr": [
    {
      "address": "0x9001",
      "progress": 0,
      "quest_id": 0,
      "noodz_name": "Noodz#29",
      "is_complete": false
    }
  ]
}`

The object should be added to questsarr. So far I have it working successfully like this, a function called "do_quests" using Supabase RPC: 

`BEGIN
  UPDATE public.quests
  set character_quests = jsonb_insert(character_quests, '{questsarr, 0}', '{"salad": "tuna salad", "tacos": "tasty"}')
  WHERE address = address_input;
END; 
`

After calling "await supabase.rpc("do_quests") it successfully adds the object to questsarr for the right user so it looks like this: 

`{
  "questsarr": [
     {
      "tacos": "tasty",
      "salad": "tuna salad"
    },
    {
      "address": "0x9001",
      "progress": 0,
      "quest_id": 0,
      "noodz_name": "Noodz#29",
      "is_complete": false
    }
  ]
}`

But how do I pass in params from the do_quest function so that they will be added to the object? 
Kinda like this: 
``BEGIN
  UPDATE public.quests
  set character_quests = jsonb_insert(character_quests, '{questsarr, 0}', '{"salad": (INSERT PARAM HERE), "tacos": (INSERT PARAM HERE) }')
  WHERE address = address_input;
END; 
``
So from the front end I could do, "await supabase.rpc.("do_quests", {tacoParamHere, saladParamHere})" and it inserts those params into the object? This methodology may not be the best lmao, but yeah, have tried a variety of diff things to get it to work but instead of elaborating on those I figured I would just ask. Thanks!!

**Comment by GaryAustin1** [comment]
There may be a way with json operators, https://www.postgresql.org/docs/9.3/functions-json.html .
But you should also be able to build up those json objects as a string variable and then use that variable in your update. 
https://www.postgresql.org/docs/current/functions-string.html

**Comment by theljwhite** [comment]
Yeah one of the attempts I did was this: 

`BEGIN
  UPDATE public.quests
  SET character_quests = jsonb_insert(character_quests, '{questsarr,0}', '{"address_input": "'|| address_input || '", "progress": ' || progress || ' ,"quest_id":' ||quest_id ||', "noodz_name": "'|| noodz_name ||'", "is_complete": "' || is_complete || '"}')
  WHERE address = address_input;
END; `

Not sure if that syntax is entirely correct as it was essentially just plugging my params in from somebody else's example that I found for a similar thing. When I use that and call from front-end "await supabase.rpc("do_quests", {address_input, progress, quest_id, noodz_name, is_complete} I get the following error:

`{code: '42883', details: null, hint: 'No function matches the given name and argument types. You might need to add explicit type casts.', message: 'function jsonb_insert(jsonb, unknown, text) does not exist'}`

Going to look into it more though, thanks for the response. Open to any other suggestions as well.
